front/src/applications/operationalStudies/components/MacroEditor/consts.ts

```diff
@@ -159,3 +159,125 @@ export const DEFAULT_PACED_TRAIN_PAYLOAD: Pick<
 };
 
 export const DEFAULT_TIME_WINDOW = new Duration({ hours: 2 });
+
+const TRAIN_CATEGORY_TO_NGE: {
+  trainCategory: TrainCategory;
+  colorRef: string;
+}[] = [
+  { trainCategory: 'HIGH_SPEED_TRAIN', colorRef: 'HST' },
+  { trainCategory: 'INTERCITY_TRAIN', colorRef: 'IC' },
+  { trainCategory: 'REGIONAL_TRAIN', colorRef: 'RE' },
+  { trainCategory: 'COMMUTER_TRAIN', colorRef: 'S' },
+  { trainCategory: 'FREIGHT_TRAIN', colorRef: 'F' },
+  { trainCategory: 'FAST_FREIGHT_TRAIN', colorRef: 'FF' },
+  { trainCategory: 'TRAM_TRAIN', colorRef: 'TT' },
+  { trainCategory: 'WORK_TRAIN', colorRef: 'W' },
+  { trainCategory: 'NIGHT_TRAIN', colorRef: 'NT' },
+  { trainCategory: 'TOURISTIC_TRAIN', colorRef: 'TO' },
+];
+
+export const OSRD_TRAINRUN_CATEGORY_MAPPING: Record<TrainCategory, TrainrunCategory> =
+  TRAIN_CATEGORY_TO_NGE.reduce(
+    (acc, { trainCategory, colorRef }, index) => {
+      acc[trainCategory] = {
+        id: index + 1,
+        order: index + 1,
+        name: trainCategory,
+        shortName: '',
+        fachCategory: 'HaltezeitUncategorized',
+        colorRef,
+        minimalTurnaroundTime: 0,
+        nodeHeadwayStop: 0,
+        nodeHeadwayNonStop: 0,
+        sectionHeadway: 0,
+      };
+      return acc;
+    },
+    {} as Record<TrainCategory, TrainrunCategory>
+  );
+OSRD_TRAINRUN_CATEGORY_MAPPING['NO_CATEGORY' as TrainCategory] = {
```

Comment:
Any reason why this needs to be a special case, and isn't included in `TRAIN_CATEGORY_TO_NGE`?

Is the reason to make it clear that the zero ID is special? If so, my suggestion above should remove the need for a special value.